### DISCUSSION TITLE
Questions about DDR with ECC

### DISCUSSION THREAD

**Discussion by gege0500** [polarfire-soc/polarfire-soc-documentation#335] [How to]
![Snipaste_2023-06-10_10-10-07](https://github.com/polarfire-soc/polarfire-soc-documentation/assets/87520674/0e8fddf1-74c7-4714-a5f2-404ea67c8bc7)

Is DDR data with ECC concatenated into 64-bit for Polarfire SoC? For example, concatenating two 32-bit data into 64 bit data for ECC.
Does the processor support 64-bit? If so, is processing 64-bit data more efficient than processing 32-bit data?

**Comment by mallynch** [comment]
Hi, Yes, the processor supports 64-bit writes. When doing a write to DDR, the DDR controller calculates the ECC bits per beat ( data width). If the data being written is less than the data width of DQ, (e.g. 8-bits being written by software to a x32 (36 inc ECC) a read modify write is carried out by the controller, so the ECC is calculated correctly before being written.
Regarding efficiency, the larger the data being written the more efficient. Data paths in the PFSoC are x64 and the DDR writes data in a burst ( one write command for multiple beats (8 or 16 burst length) ) .
